packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -228,9 +273,9 @@ const runAPI = (plugin, api, args, activity) => {
     } else {
       const result = gatsbyNode[api](...apiCallArgs)
       pluginSpan.finish()
-      return Promise.resolve(result).then(res => {
+      return Promise.resolve(result).finally(() => {
```

Comment:
previously we would not toggle `apiFinished` if plugin rejected

Reply:
actually this is wrong still - if hook is not async then `gatsby[api](...apiCallArgs)` would throw and nothing would handle this